Command.cpp

```diff
@@ -0,0 +1,220 @@
+}
+
+DeleteClassCommand::DeleteClassCommand(const std::string &name) : mName(name) {}
+
+void DeleteClassCommand::execute() const
+{
+	Command::modelInstance->removeClass(mName);
+}
+
+AddRelationshipCommand::AddRelationshipCommand(const std::string &parent,
+                                        const std::string &child,
+	const std::string &type)
+    : mParent(parent), mChild(child), mType(RelationshipFromString(type)) {}
+
+void AddRelationshipCommand::execute() const 
+{
+  Command::modelInstance->addRelationship(mParent, mChild, mType);
+}
+
+ListCommand::ListCommand() {}
+
+void ListCommand::execute() const
+{
+	Command::modelInstance->list();
+}
+
+HelpCommand::HelpCommand(){}
+
+void HelpCommand::execute() const
+{
+	std::vector<std::string> test = allCommands();
+	for (auto & i : test)
+	{
+		std::cout << i << std::endl;
+	}
+}
+HelpForCommand::HelpForCommand(const std::string &name) : mName(name) {}
+
+void HelpForCommand::execute() const
+{
+	std::cout << helpFor(mName);
+}
+
+EditRelationshipCommand::EditRelationshipCommand(const std::string& parent, const std::string& child, const std::string& type)
+	: mParent(parent), mChild(child), mType(RelationshipFromString(type)) {}
+
+void EditRelationshipCommand::execute() const
+{
+	Command::modelInstance->editRelationship(mParent, mChild, mType);
+}
+
+DeleteRelationshipCommand::DeleteRelationshipCommand(const std::string& parent, const std::string& child)
+	: mParent(parent), mChild(child) {}
+
+void DeleteRelationshipCommand::execute() const
+{
+	Command::modelInstance->removeRelationship(mParent, mChild);
+}
+
+AddFieldCommand::AddFieldCommand(const std::string& className, const std::string& fieldName, const std::string& fieldType, const std::string& fieldVisibility)
+	: mclassName(className), mfieldName(fieldName), mfieldType(fieldType), mfieldVisibility(VisibilityFromString(fieldVisibility)) {}
+
+void AddFieldCommand::execute() const
+{
+	Command::modelInstance->addField(mclassName, mfieldName, mfieldType, mfieldVisibility);
+}
+
+EditFieldCommand::EditFieldCommand(const std::string& whichAttr, const std::string& className, const std::string& fieldName, const std::string& newValue)
+	: mclassName(className), mfieldName(fieldName), mNewValue(newValue), mWhichAttr(whichAttr) {}
+
+void EditFieldCommand::execute() const
+{
+	Command::modelInstance->editField(mWhichAttr, mclassName, mfieldName, mNewValue);
+}
+
+DeleteFieldCommand::DeleteFieldCommand(const std::string& className, const std::string& fieldName)
+	: mclassName(className), mfieldName(fieldName) {}
+
+void DeleteFieldCommand::execute() const
+{
+	Command::modelInstance->deleteField(mclassName, mfieldName);
+}
+
+AddMethodCommand::AddMethodCommand(const std::string& className, const std::string& methodName, const std::string& methodType, const std::string& methodVisibility)
+	: mclassName(className), mmethodName(methodName), mmethodType(methodType), mmethodVisibility(VisibilityFromString(methodVisibility)) {}

```

Comment:
mClassName, mMethodName, mMethodVisibility

note the capitalization differences? mclass and mmethod are hard to read